utils/docker/Dockerfile.leap.15

```diff
@@ -6,7 +6,7 @@
 #
 
 # Pull base image
-ARG BASE_DISTRO=opensuse/leap:15.4
+ARG BASE_DISTRO=opensuse/leap:15.5
```

Comment:
No, it's not used for the RPM builds or it's testing.  I guess I was just thinking more about the atomicity of doing a version bump for a distro.  If nothing else it gives one something to look back at when doing it again, instead of having to hunt for multiple commits.

Now that said, it's far less than ideal that there are so many places to have to update.  I have some ideas that I have already started implementing in the GHA PR to try to make this a one-place-to-change kind of thing.  It needs to have support for specifying multiple versions to support (like EL 8.6 and 8.8 in master currently), but ultimately I want one data point that all jobs and steps to refer to to know what they should be building and what they should be testing.

The ability to dynamically build matrices in GHA is going to help with this a lot.

Reply:
I changed the gha config to specify the new version (it always defines a version) so the only impact of this line is what version a user gets if doing docker builds outside of ci, something very few people will be doing.

Your argument for atomicity of changes makes sense but does it really apply to this line given the above?

Are you saying to back out all the leap version changes or just this (unused) one?

Reply:
> Are you saying to back out all the leap version changes or just this (unused) one?

I think ultimately that is what I was asking about your thoughts on.  :smile:   I'm not terribly adamant one way or another but I am a fan of atomicity and archaeological value, just to make a future task potentially easier.  I'm fine with whichever way you want to go on this.

We will conflict on `utils/docker/Dockerfile.leap.15` though.  :frowning:

Reply:
My though process was I was asked something about recent debian and when I tested it then it didn't work so I upgraded the Dockerfile so it will work with the most recent + the most recent LTS streams (not an official policy but reasonable behaviour in my opionion).  I updated the GHA to use the most recent as this gives best coverage and is where things are likely to break.

I then did a cursory glance and the sles versions and picked up a new point version there as it seemed a good thing, other than bumping the version number there was nothing to do or I'd probably have filed a ticket and not included it.  Given the el8 status and alma/rocky discussions and the majority of Jenkins works on el8 I thought I'd leave that side to you.

There didn't appear to be a ticket for upstream OS updates that I could find.

Reply:
that's only for sles and doesn't seem appropriate for this PR.

Reply:
> that's only for sles and doesn't seem appropriate for this PR.

Not following.  Leap and SLES are like Rocky and RHEL?  We use the former of each as proxies for the latter due to the headaches of actually using the latter.  So while DAOS-14439 is about SLES on paper, in reality it is about Leap for us.

Reply:
This PR mostly changes ubuntu, and that ticket covers a lot of Jenkins work which doesn't apply to GitHub Actions.  I can use it for this ticket if you prefer.